targets/psa_switch/psa_switch.cpp

```diff
@@ -333,6 +331,17 @@ PsaSwitch::ingress_thread() {
     Deparser *deparser = this->get_deparser("ingress_deparser");
     deparser->deparse(packet.get());
 
+    // resubmit - do not move below multicast
+    Field &f_resubmit = phv->get_field("psa_ingress_output_metadata.resubmit");
+    if (f_resubmit.get_int()) {
+      // since we resubmit the copy & toss original, we update the copy's metadata
+      phv_copy->get_field("psa_ingress_parser_input_metadata.packet_path").set(PACKET_PATH_RESUBMIT);
+      phv_copy->get_field("psa_ingress_input_metadata.ingress_timestamp").set(get_ts().count());
+      BMLOG_DEBUG_PKT(*packet, "Resubmitting packet");
+      input_buffer.push_front(std::move(packet_copy));
+      continue;
+    }
```

Comment:
This resubmit check and code should, I think, be moved above the invocation of the deparser.  As Antonin mentioned, there is no functional need to invoke the deparser for resubmitted packets.  At the very least, it is inefficient and a waste of psa_switch CPU cycles.  It may also be preventing the cheaper simple_switch way of resubmitting the original packet contents from working.

Reply:
also resolved in the latest commit @jafingerhut